savant/deepstream/encoding.py

```diff
@@ -46,11 +64,20 @@ def check_encoder_is_available(parameters: Dict[str, Any]) -> bool:
             properties=converter_props,
         ),
         PipelineElement(
-            codec.value.encoder(output_frame.get('encoder')),
+            encoder,
             properties=output_frame.get('encoder_params', {}),
         ),
+        PipelineElement(
+            codec.value.parser,
+            properties={'config-interval': -1},
+        ),
+        PipelineElement(
+            'capsfilter',
+            properties={'caps': output_caps},
+        ),
         PipelineElement('fakesink'),
     ]
+
     last_gst_element = None
     for element in elements:
         if element.element == 'capsfilter':
```

Comment:
It will do GstElementFactory, no need to check `if element.element == 'capsfilter'`.

Reply:
Removed unnecessary check.

Reply:
Cannot use `GstElementFactory.create()` since it creates `videotestsrc` as a bin and it doesn't link to the `capsfiler`.